instrumentation/grape/lib/opentelemetry/instrumentation/grape/subscriber.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Grape
+      # Manages all subscriptions, both for custom subscribers and built-in notifications
+      class Subscriber
+        class << self
+          # Subscribe to all custom and built-in notifications (except those specified in the :ignored_events configs)
+          def subscribe
+            subscribe_to_custom_notifications
+            subscribe_to_built_in_notifications
+          end
+
+          private
+
+          SUBSCRIPTIONS = {
+            endpoint_render: 'endpoint_render.grape',
+            endpoint_run_filters: 'endpoint_run_filters.grape',
+            format_response: 'format_response.grape'
+          }.freeze
+
+          CUSTOM_SUBSCRIPTIONS = {
+            endpoint_run: 'OpenTelemetry::Instrumentation::Grape::CustomSubscribers::EndpointRun'
+          }.freeze
+
+          def subscribe_to_custom_notifications
+            CUSTOM_SUBSCRIPTIONS.each do |event, klass|
+              custom_subscriber = Object.const_get(klass).new
```

Comment:
oof I will never get used to ruby metaprogramming 🤦‍♂️

Reply:
Yeah, I'm not a big fan either 😅